stackit/internal/services/iaas/networkarea/resource.go

```diff
@@ -0,0 +1,678 @@
+			config.WithCustomAuth(providerData.RoundTripper),
+			config.WithEndpoint(providerData.IaaSCustomEndpoint),
+		)
+	} else {
+		apiClient, err = iaas.NewAPIClient(
+			config.WithCustomAuth(providerData.RoundTripper),
+			config.WithRegion(providerData.Region),
+		)
+	}
+
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error configuring API client", fmt.Sprintf("Configuring client: %v. This is an error related to the provider configuration, not to the resource configuration", err))
+		return
+	}
+
+	r.client = apiClient
+	tflog.Info(ctx, "IaaS client configured")
+}
+
+// Schema defines the schema for the resource.
+func (r *networkAreaResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Network area resource schema. Must have a `region` specified in the provider configuration.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "Terraform's internal resource ID. It is structured as \"`organization_id`,`network_area_id`\".",
+				MarkdownDescription: features.AddBetaDescription("Network area resource schema. Must have a `region` specified in the provider configuration."),
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"organization_id": schema.StringAttribute{
+				Description: "STACKIT organization ID to which the network area is associated.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"network_area_id": schema.StringAttribute{
+				Description: "The network area ID.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the network area.",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					stringvalidator.LengthAtMost(63),
+				},
+			},
+			"project_count": schema.Int64Attribute{
+				Description: "The amount of projects currently referencing this area.",
+				Computed:    true,
+				Validators: []validator.Int64{
+					int64validator.AtLeast(0),
+				},
+			},
+			"default_nameservers": schema.ListAttribute{
+				Description: "List of DNS Servers/Nameservers.",
+				Optional:    true,
+				ElementType: types.StringType,
+			},
+			"network_ranges": schema.SetAttribute{
+				Description: "List of Network ranges.",
+				ElementType: types.StringType,
+				Required:    true,
+				Validators: []validator.Set{
+					setvalidator.SizeAtLeast(1),
+					setvalidator.SizeAtMost(64),
+				},
+			},
```

Comment:
The network range field in the API is a list of objects, each with a `prefix` and a `networkRangeId` fields. Shouldn't we replicate this here in the provider instead o making it a list of just the string prefixes?

Reply:
We could also add a field for the computed `networkRangeId` field, right? (it is returned in the response of the [`CreateArea`](https://docs.api.stackit.cloud/documentation/iaas/version/v1#tag/Area/operation/v1CreateArea) endpoint)

Reply:
thx for the catch, I had it in datasource but not in resource :(